shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/controler/StudyCardApiController.java

```diff
@@ -50,177 +50,184 @@
-			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-		}
-		return new ResponseEntity<>(studyCards, HttpStatus.OK);
-	}
-
-	@Override
-	public ResponseEntity<Void> updateStudyCard(
-			@ApiParam(value = "id of the study card", required = true) @PathVariable("studyCardId") Long studyCardId,
-			@ApiParam(value = "study card to update", required = true) @RequestBody StudyCard studyCard,
-			final BindingResult result) throws RestServiceException {
-		validate(studyCard, result);
-		try {
-			studyCardService.update(studyCard);
-			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-		} catch (EntityNotFoundException e) {
-			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-		} catch (MicroServiceCommunicationException e) {
-			throw new RestServiceException(
-					new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), MICROSERVICE_COMMUNICATION_ERROR, null));
-		}
-	}
-	
-	@Override
-	public ResponseEntity<List<DicomTag>> findDicomTags() throws RestServiceException {
-		Field[] declaredFields = Tag.class.getDeclaredFields();
-		List<DicomTag> dicomTags = new ArrayList<DicomTag>();
-		try {
-			for (Field field : declaredFields) {
-			    if (java.lang.reflect.Modifier.isStatic(field.getModifiers())) {
-			    	if (field.getType().getName() == "int")
-						dicomTags.add(new DicomTag(field.getInt(null), field.getName()));
-			    	// longs actually code a date and a time, see Tag.class
-			    	if (field.getType().getName() == "long") {
-			    		String name = field.getName().replace("DateAndTime", "");
-			    		String hexStr = String.format("%016X", field.getLong(null));
-			    		String dateStr = hexStr.substring(0, 8);
-			    		String timeStr = hexStr.substring(8);
-			    		dicomTags.add(new DicomTag(Integer.parseInt(dateStr, 16), name + "Date"));
-			    		dicomTags.add(new DicomTag(Integer.parseInt(timeStr, 16), name + "Time"));
-			    	}
-			    }
-			}
-		} catch (IllegalArgumentException | IllegalAccessException e) {
-			throw new RestServiceException(new ErrorModel(HttpStatus.INTERNAL_SERVER_ERROR.value(), "Cannot parse the dcm4che lib Tag class static fields", e));
-		}
-		return new ResponseEntity<>(dicomTags, HttpStatus.OK);
-	}
-
-	/**
-	 * Validate a studyCard
-	 * 
-	 * @param studyCard
-	 * @param result
-	 * @throws RestServiceException
-	 */
-	protected void validate(StudyCard studyCard, BindingResult result) throws RestServiceException {
-		final FieldErrorMap errors = new FieldErrorMap()
-				.add(new FieldErrorMap(result))
-				.add(uniqueConstraintManager.validate(studyCard));
-		if (!errors.isEmpty()) {
-			ErrorModel error = new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Bad arguments", new ErrorDetails(errors));
-			throw new RestServiceException(error);
-		} 
-	}
-	
-	@Override
-	public ResponseEntity<Void> applyStudyCard(
-			@ApiParam(value = "study card id and dataset ids", required = true) @RequestBody StudyCardApply studyCardApplyObject) throws RestServiceException, PacsException {
```

Comment:
this should stay

(there is a compilation error yet : 2023-05-05T11:22:33.1916613Z [ERROR] /home/runner/work/shanoir-ng/shanoir-ng/shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/controler/StudyCardApiController.java:[229,45] unreported exception org.shanoir.ng.shared.exception.PacsException; must be caught or declared to be thrown)